From ea27f256fb40110b2a36e57bf2106069c28a9c4a Mon Sep 17 00:00:00 2001
From: Martin Lang <martin.lang@mpsd.mpg.de>
Date: Wed, 26 Nov 2025 12:27:08 +0100
Subject: [PATCH] Replace blessings with blessed

- blessed is an actively maintained fork of blessings
- it also supports Windows (unlike blessings)

I have picked the oldest release of blessed that has been released after
the release of blessings 1.7 as lower limit.
---
 README.md           | 2 +-
 ontopy/colortest.py | 2 +-
 requirements.txt    | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/README.md b/README.md
index 767ff1c..9e614df 100644
--- a/README.md
+++ b/README.md
@@ -219,7 +219,7 @@ It has mainly been developed by [SINTEF](https://www.sintef.no/), specifically:
 [pdfLaTeX]: https://www.latex-project.org/
 [graphviz_python]: https://pypi.org/project/graphviz
 [PyYAML]: https://pypi.org/project/PyYAML/
-[blessings]: https://pypi.org/project/blessings/
+[blessed]: https://pypi.org/project/blessed/
 [Pygments]: https://pypi.org/project/Pygments/
 [rdflib]: https://pypi.org/project/rdflib/
 [FaCT++]: http://owl.cs.manchester.ac.uk/tools/fact/
diff --git a/ontopy/colortest.py b/ontopy/colortest.py
index 0be8fe6..cbdaa7e 100644
--- a/ontopy/colortest.py
+++ b/ontopy/colortest.py
@@ -11,7 +11,7 @@ from unittest.result import TestResult
 from unittest.runner import TextTestRunner
 from unittest.util import strclass
 
-from blessings import Terminal
+from blessed import Terminal
 from pygments import formatters, highlight
 from pygments.lexers import (  # pylint: disable=no-name-in-module
     Python3TracebackLexer as Lexer,
diff --git a/requirements.txt b/requirements.txt
index 17691a4..75d67e8 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-blessings>=1.7,<2  # core functionality ontology check
+blessed>=1.17  # core functionality ontology check
 defusedxml>=0.7.1,<1  # core functionality parsing
 Owlready2>=0.28,!=0.32,!=0.34,<0.49  # core functionality ontology
 packaging>=21.0,<26
